pkg/pfconfig/shared/shared.go

```diff
@@ -108,10 +105,80 @@ func (c SX1301Config) MarshalJSON() ([]byte, error) {
 	return json.Marshal(m)
 }
 
-// UnmarshalJSON implements json.Unmarshaler. It currently just errors because
-// unmarshaling is not supported.
-func (c *SX1301Config) UnmarshalJSON(_ []byte) error {
-	return errUnmarshalNotImplemented
+// UnmarshalJSON implements json.Unmarshaler.
+func (c *SX1301Config) UnmarshalJSON(msg []byte) error {
+	var root map[string]json.RawMessage
+	if err := json.Unmarshal(msg, &root); err != nil {
+		return err
+	}
+	radioMap, txLutMap, chanMap := make(map[int]RFConfig), make(map[int]TxLUTConfig), make(map[int]IFConfig)
+	for key, value := range root {
+		switch {
+		case key == "lorawan_public":
+			if err := json.Unmarshal(value, &c.LoRaWANPublic); err != nil {
+				return err
+			}
+		case key == "antenna_gain":
+			if err := json.Unmarshal(value, &c.AntennaGain); err != nil {
+				return err
+			}
+		case key == "clksrc":
+			if err := json.Unmarshal(value, &c.ClockSource); err != nil {
+				return err
+			}
+		case key == "lbt_cfg":
+			if err := json.Unmarshal(value, &c.LBTConfig); err != nil {
+				return err
+			}
+		case key == "chan_Lora_std":
+			if err := json.Unmarshal(value, &c.LoRaStandardChannel); err != nil {
+				return err
+			}
+		case key == "chan_FSK":
+			if err := json.Unmarshal(value, &c.FSKChannel); err != nil {
+				return err
+			}
+		case strings.HasPrefix(key, "chan_multiSF_"):
+			var channel IFConfig
+			if err := json.Unmarshal(value, &channel); err != nil {
+				return err
+			}
+			var index int
+			if _, err := fmt.Sscanf(key, "chan_multiSF_%d", &index); err == nil {
```

Comment:
We could be a bit more strict and return this `err` if it's non-nil

Reply:
ok done. Re-requesting review since it won't let me merge. 🙏